ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LocalRootPeers.hs

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+module Ouroboros.Network.PeerSelection.LocalRootPeers (OutboundConnectionsState (..)) where
+
+import GHC.Generics
+import NoThunks.Class
+
+data OutboundConnectionsState =
+    TrustedStateWithExternalPeers
+    -- ^
+    -- * /in praos mode/: connected only to trusted local
+    --   peers and at least one bootstrap peer or public root;
+    -- * /in genesis mode/: meeting target of active big ledger peers;
+    -- * or it is in promiscuous mode (see below).
+    --
+    -- A node is classified as a non promiscuous node if it is a hidden relay or
+    -- a BPa, e.g. if it is configured such that it can only have a chance to be
+    -- connected to local roots. This is true in one of the two ways:
+    --
+    -- * it's configured with `DontUseBootstrapPeers` and `DontUseLedgerPeers`;
```

Comment:
and with peersharing disabled.

Reply:
I forgot to update the comment.  I added `PromiscuousMode` type in `Governor.Types` where this comment lives now.